homeassistant/components/velux/__init__.py

```diff
@@ -53,15 +53,9 @@ def __init__(self, hass, domain_config):
     def setup(self):
         """Velux component setup."""
 
-        async def on_hass_stop(event):
-            """Close connection when hass stops."""
-            _LOGGER.debug("Velux interface terminated")
-            await self.pyvlx.disconnect()
```

Comment:
It seems weird that we would not cleanly disconnect. Why is an automation even necessary to reboot. Seems like we're now asking each used to solve the same problem

Reply:
After some hours of operation the KLF200 Gateway refuses new connections. If HA restarts, the connection build-up run into timeout during SSL Handshake. So if I start a reboot of the KLF200 before HA shutdown I can establish a new connection after HA is rebooted. Finall it's just a workaround. A disconnect function is considered in the destructor of [pyvlx.connection](https://github.com/Julius2342/pyvlx/blob/77b88b6ed1e5b4d750fdfd0a01abc78b9479f706/pyvlx/connection.py#L74), so I think it will not change much to remove this functionality on HA component side.

Reply:
Yes of course it would be much simpler for the user if we hardcode that, so they will not need to create an automation for that. Only point to consider is that KLF200 starts a wifi access point after each reboot where you can perform configurations. It's usually protected by a password if the user does not remove that. In the standard configuration this access point will be closed after some minutes. Finally my safety concerns are the reason not to hardcode this automation.

Reply:
@pawlizio : IMO all explicit disconnects are preferred bc then the disconnects happen in a controlled way. I guess als hass-shutdown process waits untill the disconnect is finished.

But yes, this buggy behavior of the KLF 200 gateway is annoying. There is another PR from you #42773 - where you implemented an explicit service to reboot the gateway from time to time. Isn't this the better approach?

Reply:
> Only point to consider is that KLF200 starts a wifi access point after each reboot where you can perform configurations. [...] Finally my safety concerns are the reason not to hardcode this automation.

Valid point m-/ .

Reply:
The reboot service is already available in the component and yes you can start them periodically. But my preferred way was to initiate the reboot in parallel to a HA restart. Main concern was to reduce the amount of unnecessary KLF reboots and the corresponding opening of the Wifi Access Point to a bare minimum.